app/services/parameters/validator.rb

```diff
@@ -0,0 +1,54 @@
+module Foreman
+  module Parameters
+    class Validator
+      KEY_DELM = ","
+
+      def initialize(item, options = {})
+        @item, @options = item, options
+      end
+
+      def validate!
+        case @options[:type].to_s.to_sym
```

Comment:
This is a bit different to the caster API, which requires that a symbol is passed in - it'd be good if they had consistent behaviour.